doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
-   functionality you want to test and add an empty ``__init__.py`` file to
+   machinery you want to test and add an empty ``__init__.py`` file to
    it.
-#. Inside that directory, create a new python file prefixed with ``test_`` and
+#. Inside this new ``tests/`` directory, create a new python file prefixed with ``test_`` and
    including the name of the functionality.
-#. Inside that file, create one or more routines prefixed with ``test_`` that
-   accept no arguments. The test function should do some work that tests some
+#. Inside this new ``test_`` file, create one or more routines prefixed with ``test_`` that
+   accept no arguments.
+#. Each test function should do some work that tests some
    functionality and should also verify that the results are correct using
    assert statements or functions.
-#. Tests can ``yield`` a tuple of the form ``function``, ``argument_one``,
-   ``argument_two``, etc.  For example ``yield my_test, 'banana', 2.0`` would be
-   captured by nose and the ``my_test`` function will be run with the provided
-   arguments.
-#. Use ``fake_random_ds`` to test on datasets, and be sure to test for
-   several combinations of ``nproc``, so that domain decomposition can be
+#. If a dataset is needed, use ``fake_random_ds`` and be sure to test for
+   several combinations of ``nproc`` so that domain decomposition can be
    tested as well.
-#. Test multiple combinations of options by using the
-   :func:`~yt.testing.expand_keywords` function, which will enable much
-   easier iteration over options.
+#. To iterate over multiple options, or combinations of options,
+   use the :ref:`@pytest.mark.parametrize <@pytest.mark.parametrize>` decorator.
 
 For an example of how to write unit tests, look at the file
 ``yt/data_objects/tests/test_covering_grid.py``, which covers a great deal of
 functionality.
 
-Debugging failing tests
+Debugging Failing Tests
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-When writing new tests, often one exposes bugs or writes a test incorrectly,
+When writing new tests, one often exposes bugs or writes a test incorrectly,
 causing an exception to be raised or a failed test. To help debug issues like
-this, ``nose`` can drop into a debugger whenever a test fails or raises an
-exception. This can be accomplished by passing ``--pdb`` and ``--pdb-failures``
-to the ``nosetests`` executable. These options will drop into the pdb debugger
-whenever an error is raised or a failure happens, respectively. Inside the
+this, ``pytest`` can drop into a debugger whenever a test fails or raises an
+exception. This can be accomplished by passing ``--pdb``
+to the ``pytest`` executable. Inside the
 debugger you can interactively print out variables and go up and down the call
 stack to determine the context for your failure or error.
 
 .. code-block:: bash
 
-    nosetests --pdb --pdb-failures
+    pytest --pdb
 
 In addition, one can debug more crudely using print statements. To do this,
```

Comment:
Before talking about print statements, I would link to the pytest documentation on dropping into a debugger. The set_trace() and breakpoint() stuff is useful and there's no need for us to repeat it here but it'd be nice to have a convenient link from our docs.